### PR TITLE
FAT-2665/C350647

### PR DESCRIPTION
**Pull request by re-jas**
[Verify automated patron block 'Maximum number of items charged out' removed after charged item returned](https://foliotest.testrail.io/index.php?/cases/view/350647)

### CODE REVIEW

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  });
+
+  it(
+    'C350647 Verify automated patron block "Maximum number of items charged out" removed after charged item returned (vega)',
+    { tags: [TestTypes.criticalPath, devTeams.vega] },
+    () => {
+      cy.visit(SettingsMenu.conditionsPath);
+      Conditions.waitLoading();
+      Conditions.select('Maximum number of items charged out');
+      Conditions.setConditionState('Patron is Blocked');
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:177]: Please change the message with the text: You have reached the maximum number of items you can check out as set by patron group.

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+      { instanceTitle: `Instance ${getRandomPostfix()}` },
+      { instanceTitle: `Instance ${getRandomPostfix()}` },
+      { instanceTitle: `Instance ${getRandomPostfix()}` },
+      { instanceTitle: `Instance ${getRandomPostfix()}` },
+      { instanceTitle: `Instance ${getRandomPostfix()}` },
+    ],
+  };
+  const testData = {
+    userServicePoint: ServicePoints.getDefaultServicePointWithPickUpLocation(
+      'autotest receive notice triggers',
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:47]: please rename the servicepoint's name to "autotest charged out limit"

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        });
+        cy.getHoldingTypes({ limit: 1 }).then((res) => {
+          testData.holdingTypeId = res[0].id;
+        });
+        cy.getLoanTypes({ limit: 1 }).then((res) => {
+          testData.loanTypeId = res[0].id;
+        });
+        cy.getMaterialTypes({ limit: 1 }).then((res) => {
+          testData.materialTypeId = res.id;
+          testData.materialTypeName = res.name;
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:73]: Do you use this property?

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 1 above)
Comment (penfriend) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:177]: Also save the text of this message to the variable and pass the variable as the argument to the setConditionState function. The var can have the name: "checkedOutBlockMessage"

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+      Conditions.waitLoading();
+      Conditions.select('Maximum number of items charged out');
+      Conditions.setConditionState('Patron is Blocked');
+      cy.visit(SettingsMenu.limitsPath);
+      Limits.selectGroup(patronGroup.name);
+      Limits.setMaximumNumberOfItemsChargedout('4');
+
+      cy.visit(TopMenu.usersPath);
+      UsersSearchPane.searchByKeywords(userData.barcode);
+      Users.checkIsPatronBlocked('Patron is Blocked', 'Borrowing, Renewals, Requests');
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:184]: pass the first argument as the variable checkedOutBlockMessage

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+      cy.visit(TopMenu.checkInPath);
+      const itemForCheckIn = itemsData.itemsWithSeparateInstance[0];
+      CheckInActions.checkInItemGui(itemForCheckIn.barcode);
+      CheckInActions.verifyLastCheckInItem(itemForCheckIn.barcode);
+
+      cy.visit(TopMenu.usersPath);
+      UsersSearchPane.searchByKeywords(userData.barcode);
+      Users.checkIsPatronBlocked('Patron is Blocked', 'Borrowing');
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:193]: pass the first argument as the variable checkedOutBlockMessage

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+      cy.visit(SettingsMenu.limitsPath);
+      Limits.selectGroup(patronGroup.name);
+      Limits.setMaximumNumberOfItemsChargedout('5');
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:197]: ChargedOut or CheckedOut, choose the one

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+      cy.visit(TopMenu.usersPath);
+      UsersSearchPane.searchByKeywords(userData.barcode);
+      Users.checkIsNotPatronBlocked();
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:201]: checkPatronIsNotBlocked

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+describe('Patron Block: Maximum number of items charged out', () => {
+  const patronGroup = {
+    name: 'groupToPatronBlock' + getRandomPostfix(),
+  };
+  const userData = {
+    personal: {
+      lastname: null,
+    },
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:34]: do you need property "personal"?

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js [hunk 9]
```diff
@@ -0,0 +1,204 @@
+import uuid from 'uuid';
```
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js:1]: can you please clarify the file name. the test case doesn't verify the returning anything. we need to rename the file

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 7 above)
Comment (penfriend) [inline cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: before this check you need to wait until user's detail page is loaded. Please add this waiter

**Review by IhorBohdan — APPROVED** [review]

File: cypress/e2e/settings/users/patron-block/charged.out.removed.cy.js [hunk 10]
```diff
@@ -0,0 +1,200 @@
+import uuid from 'uuid';
```
Comment (MaksimTerekhov) [inline cypress/e2e/settings/users/patron-block/charged.out.removed.cy.js:1]: let's remove file name to more understandable
patron-block-maximum-items-charged-out-removed.cy.js
thx

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 2 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Done

File: cypress/e2e/checkout/checkout.patron.block.cy.js [hunk 11]
```diff
@@ -0,0 +1,204 @@
+import uuid from 'uuid';
+import moment from 'moment';
```
[the remark this answers is not in the document]
Comment (re-jas) [reply cypress/e2e/checkout/checkout.patron.block.cy.js]: Done

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 1 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Done

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 3 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Removed

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 4 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Done

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 5 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Done

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 6 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Сorrected

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 7 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Сorrected

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 8 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Removed

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 9 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Done

File: cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js (hunk 7 above)
Comment (re-jas) [reply cypress/e2e/settings/users/patron-block/maximum.charged.out.returned.cy.js]: Added

**Review by penfriend — APPROVED** [review]

**Pull request merged**